src/compat/array/filter.ts

```diff
@@ -165,6 +164,22 @@ export function filter<T>(
 
   switch (typeof predicate) {
     case 'function': {
+      if (!Array.isArray(source)) {
+        const result: T[] = [];
+        const entries: any[] = Object.entries(source);
+
+        for (let i = 0; i < entries.length; i++) {
+          const entry = entries[i];
+          const [key, value] = entry;
```

Comment:
```suggestion
          const key = entry[0];
          const value = entry[1];
```
We're accessing `key` and `value` with indices for performance reasons.